v3/lints/rfc/lint_name_constraint_not_fqdn_test.go

```diff
@@ -0,0 +1,65 @@
+package rfc
+
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"testing"
+
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/test"
+)
+
+func TestUriNameConstraintsFqdn(t *testing.T) {
+	testCases := []struct {
+		Name           string
+		Filename       string
+		ExpectedResult lint.LintStatus
+	}{
+		{
+			Name:           "TestBeginsWithPeridFQDN",
+			Filename:       "beginsWithPeriodConstraintFQDN.pem",
+			ExpectedResult: lint.Pass,
+		},
+		{
+			Name:           "TestIpAddressNotFQDN",
+			Filename:       "ipAddressConstraintNotFQDN.pem",
+			ExpectedResult: lint.Error,
+		},
+		{
+			Name:           "TestOnlyHostFQDN",
+			Filename:       "onlyHostConstraintFQDN.pem",
+			ExpectedResult: lint.Pass,
+		},
+		{
+			Name:           "TestNoAuthorityNotFQDN",
+			Filename:       "noAuthorityConstraintNotFQDN.pem",
+			ExpectedResult: lint.Error,
+		},
+		{
+			Name:           "TestMultipleWrongContstraintsNotFQDN",
+			Filename:       "multipleWrongConstraints.pem",
+			ExpectedResult: lint.Error,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			result := test.TestLint("e_name_constraint_not_fqdn", tc.Filename)
+			if result.Status != tc.ExpectedResult {
```

Comment:
It would be nice to also get test coverage of the logic you wrote in `returnErrorString`. What do you think about extending the `testCases` to also include the expected details message and verifying that as well as the `result.Status`. I think that would help us get confidence that the string manipulation in `returnErrorString` is correct.

Reply:
Okay! I will include a test for every case then. I actually already tested most of them I'm not sure why I decided against including those in my last commit. :'D

Reply:
I have already had test files for:
1. 1 inclusion constraint
2.  multiple inclusion constraints
I added the following cases yesterday:
1. 1 exclusion constraint
2. 1 inclusion constraint and multiple exclusion constraints
3.  more than 1 constraint of both types
4. 1 constraint of both types

And now I'm realizing that I misread your comment yesterday. :'D I will include the error messages in my next commit. Sorry for this.

Reply:
Hi @ZsofiaTomicsko,

Did you end up pushing another commit for this? I see that L83 is logging the details message but I don't see the expected details message reflected in the `testCases`, or an assertion that the details message in `result` matches the expected one from the test case